internal/js_library/js_library.bzl

```diff
@@ -49,96 +53,124 @@ def write_amd_names_shim(actions, amd_names_shim, targets):
                 amd_names_shim_content += "define(\"%s\", function() { return %s });\n" % n
     actions.write(amd_names_shim, amd_names_shim_content)
 
-def _impl(ctx):
-    files = []
-    typings = []
-    js_files = []
+def _js_library_impl(ctx):
+    direct_sources = depset(ctx.files.srcs)
+    direct_named_module_sources = depset(ctx.files.named_module_srcs)
+    sources_depsets = [direct_sources]
+    named_module_sources_depsets = [direct_named_module_sources]
 
+    include_npm_package_info = False
     for src in ctx.files.srcs:
-        if src.is_source and not src.path.startswith("external/"):
-            dst = ctx.actions.declare_file(src.basename, sibling = src)
-            if ctx.attr.is_windows:
-                copy_cmd(ctx, src, dst)
-            else:
-                copy_bash(ctx, src, dst)
-            if dst.basename.endswith(".d.ts"):
-                typings.append(dst)
-            else:
-                files.append(dst)
-        elif src.basename.endswith(".d.ts"):
-            typings.append(src)
-        else:
-            files.append(src)
-
-    for p in files:
-        if p.basename.endswith(".js") or p.basename.endswith(".js.map") or p.basename.endswith(".json"):
-            js_files.append(p)
-
-    files_depset = depset(files)
+        if src.is_source and src.path.startswith("external/"):
+            include_npm_package_info = True
```

Comment:
this merits a comment about why it's done this way

Reply:
included